### PR TITLE
CNS-131: added events for unstaking

### PR DESCRIPTION
**Pull request by oren-lava**
* **Please check if the PR fulfills these requirements**
- [ v ] Commit message follows the Contribution Guidelines
- [ v ] Tests ran locally and added/modified if needed
- [ v ] Docs have been added/updated, if applicable
- [ v ] If applicable - JIRA ticket ID was added

* **What kind of change does this PR introduce?** (Bug fix, feature, unit tests, docs update, ...)
feature - Add events messages for unstaking

* **What is the current behavior?** (You can also link to an open issue here)
The only unstake-related event is when the unstaker's money is returned to him. There are no events when the unstake is done or for when the unstake is automatic due to unresponsive provider / vote fraud

* **What is the new behavior (if this is a feature change)?**
All unstake-related scenarios have events

* **Please describe what manual tests you ran, if applicable**
There was no need to create a new test since I didn't change any functionality, only added "log" events. I've ran all the existing tests and they all passed.

* **Other information**:

### CODE REVIEW

File: x/pairing/keeper/stakeModify.go [hunk 1]
```diff
@@ -29,7 +30,8 @@ func (k Keeper) BurnClientStake(ctx sdk.Context, chainID string, clientAddressTo
 
 		if clientEntry.Stake.IsLT(k.MinStakeClient(ctx)) {
 			// if user doesn't have enough stake to stay staked, we will unstake him now
-			logger.Info("unstaking client", clientEntry.Address, "insufficient funds to stay staked", clientEntry.Stake)
+			details := map[string]string{"stake_entry": clientEntry.String()}
+			utils.LogLavaEvent(ctx, logger, types.ConsumerInsufficientFundsToStayStakedEventName, details, " insufficient funds to stay staked.")
```
Comment (Yaroms) [inline x/pairing/keeper/stakeModify.go:34]: there is an extra space at the beginning of each description

File: x/pairing/keeper/unstaking.go [hunk 2]
```diff
@@ -52,13 +52,22 @@ func (k Keeper) UnstakeEntry(ctx sdk.Context, provider bool, chainID string, cre
 	if existingEntry.Deadline < holdBlocks {
 		existingEntry.Deadline = holdBlocks
 	}
+	details := map[string]string{"stake_entry": existingEntry.String()}
+	utils.LogLavaEvent(ctx, logger, types.UnstakeCommitNewEventName(provider), details, " Appending new unstake entry (removing stake entry).")
```
Comment (Yaroms) [inline x/pairing/keeper/unstaking.go:56]: you added an event for each time you use this function, we dont need another one inside, you can maybe send one event from this function and add description string as a parameter for the function

File: x/pairing/keeper/unstaking.go [hunk 3]
```diff
@@ -52,13 +52,22 @@ func (k Keeper) UnstakeEntry(ctx sdk.Context, provider bool, chainID string, cre
...
 func (k Keeper) CheckUnstakingForCommit(ctx sdk.Context) error {
 	// this pops all the entries that had their deadline pass
 	unstakingEntriesToCredit := k.epochStorageKeeper.PopUnstakeEntries(ctx, epochstoragetypes.ProviderKey, uint64(ctx.BlockHeight()))
+
 	if unstakingEntriesToCredit != nil {
+		details := map[string]string{"unstake_entities_type": epochstoragetypes.ProviderKey}
+		for i, unstakeEntry := range unstakingEntriesToCredit {
+			details["unstake_entry_"+strconv.Itoa(i)] = unstakeEntry.String()
+		}
+		utils.LogLavaEvent(ctx, k.Logger(ctx), types.UnstakeCommitNewEventName(true), details, " Crediting "+epochstoragetypes.ProviderKey+" unstake entries.")
```
Comment (Yaroms) [inline x/pairing/keeper/unstaking.go:69]: there is already an event inside k.creditUnstakingEntries

Comment (Yaroms) [reply x/pairing/keeper/unstaking.go:69]: revert what you added

File: x/pairing/keeper/unstaking.go [hunk 4]
```diff
@@ -67,6 +76,12 @@ func (k Keeper) CheckUnstakingForCommit(ctx sdk.Context) error {
 	// no providers entries to handle, check clients
 	unstakingEntriesToCredit = k.epochStorageKeeper.PopUnstakeEntries(ctx, epochstoragetypes.ClientKey, uint64(ctx.BlockHeight()))
 	if unstakingEntriesToCredit != nil {
+		details := map[string]string{"unstake_entities_type": epochstoragetypes.ClientKey}
+		for i, unstakeEntry := range unstakingEntriesToCredit {
+			details["unstake_entry_"+strconv.Itoa(i)] = unstakeEntry.String()
+		}
+		utils.LogLavaEvent(ctx, k.Logger(ctx), types.UnstakeCommitNewEventName(false), details, " Crediting "+epochstoragetypes.ClientKey+" unstake entries.")
```
Comment (Yaroms) [inline x/pairing/keeper/unstaking.go:83]: same here

**Review by Yaroms — CHANGES_REQUESTED** [review]

File: x/pairing/keeper/stakeModify.go (hunk 1 above)
Comment (oren-lava) [reply x/pairing/keeper/stakeModify.go:34]: I saw that when it actually prints the message, it prints it with no space after the end of the previous sentence, so these extra spaces are on purpose.
Remove them?

Comment (oren-lava) [reply x/pairing/keeper/stakeModify.go:34]: after talking with Yarom, I'll keep it

File: x/pairing/keeper/unstaking.go (hunk 3 above)
Comment (oren-lava) [reply x/pairing/keeper/unstaking.go:69]: done

File: x/pairing/keeper/unstaking.go (hunk 4 above)
Comment (oren-lava) [reply x/pairing/keeper/unstaking.go:83]: done

File: x/pairing/keeper/unstaking.go (hunk 2 above)
Comment (oren-lava) [reply x/pairing/keeper/unstaking.go]: done

File: x/conflict/keeper/vote.go [hunk 5]
```diff
@@ -165,9 +165,7 @@ func (k Keeper) HandleAndCloseVote(ctx sdk.Context, conflictVote types.ConflictV
 						utils.LavaError(ctx, logger, "slash_failed_vote", map[string]string{"error": err.Error()}, "slashing failed at vote conflict")
 					}
 
-					details := map[string]string{"fraud voter address": vote.Address, "fraud voter chainID": conflictVote.ChainID}
-					utils.LogLavaEvent(ctx, logger, types.ConflictUnstakeFraudVoterEventName, details, " unstaking fraud voter.")
-					err = k.pairingKeeper.UnstakeEntry(ctx, true, conflictVote.ChainID, vote.Address)
+					err = k.pairingKeeper.UnstakeEntry(ctx, true, conflictVote.ChainID, vote.Address, " unstaking fraud voter.")
```
Comment (Yaroms) [inline x/conflict/keeper/vote.go:168]: rewrite - fraud provider found in conflict detection

**Review by Yaroms — CHANGES_REQUESTED** [review]

File: x/conflict/keeper/vote.go (hunk 5 above)
Comment (oren-lava) [reply x/conflict/keeper/vote.go:168]: done

File: x/pairing/keeper/msg_server_unstake_client.go [hunk 6]
```diff
@@ -9,6 +9,6 @@ import (
 
 func (k msgServer) UnstakeClient(goCtx context.Context, msg *types.MsgUnstakeClient) (*types.MsgUnstakeClientResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	err := k.Keeper.UnstakeEntry(ctx, false, msg.ChainID, msg.Creator)
+	err := k.Keeper.UnstakeEntry(ctx, false, msg.ChainID, msg.Creator, " Appending new client unstake entry (removing stake entry).")
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/msg_server_unstake_client.go:12]: done

File: x/pairing/keeper/msg_server_unstake_provider.go [hunk 7]
```diff
@@ -9,6 +9,7 @@ import (
 
 func (k msgServer) UnstakeProvider(goCtx context.Context, msg *types.MsgUnstakeProvider) (*types.MsgUnstakeProviderResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	err := k.Keeper.UnstakeEntry(ctx, true, msg.ChainID, msg.Creator)
+
+	err := k.Keeper.UnstakeEntry(ctx, true, msg.ChainID, msg.Creator, " Appending new provider unstake entry (removing stake entry).")
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/msg_server_unstake_provider.go:13]: done

File: x/pairing/keeper/stakeModify.go [hunk 8]
```diff
@@ -30,10 +28,8 @@ func (k Keeper) BurnClientStake(ctx sdk.Context, chainID string, clientAddressTo
 
 		if clientEntry.Stake.IsLT(k.MinStakeClient(ctx)) {
 			// if user doesn't have enough stake to stay staked, we will unstake him now
-			details := map[string]string{"stake_entry": clientEntry.String()}
-			utils.LogLavaEvent(ctx, logger, types.ConsumerInsufficientFundsToStayStakedEventName, details, " insufficient funds to stay staked.")
 			// err := k.UnstakeUser(ctx, chainID, specStakeStorage.StakeStorage.StakedUsers[idx].Index, types.BlockNum{Num: 0})
-			err := k.UnstakeEntry(ctx, false, chainID, clientEntry.Address)
+			err := k.UnstakeEntry(ctx, false, chainID, clientEntry.Address, " insufficient funds to stay staked.")
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/stakeModify.go:32]: done

File: x/conflict/keeper/vote.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/conflict/keeper/vote.go:168]: done

**Review by Yaroms — APPROVED** [review]

File: x/pairing/keeper/unstaking.go [hunk 9]
```diff
@@ -52,12 +52,15 @@ func (k Keeper) UnstakeEntry(ctx sdk.Context, provider bool, chainID string, cre
 	if existingEntry.Deadline < holdBlocks {
 		existingEntry.Deadline = holdBlocks
 	}
+	details := map[string]string{"stake_entry": existingEntry.String()}
```
Comment (Yaroms) [inline x/pairing/keeper/unstaking.go:55]: did you check how it looks?

Comment (oren-lava) [reply x/pairing/keeper/unstaking.go]: changed it to be more readable

**Review by Yaroms — APPROVED** [review]

**Pull request merged**